src/thx/Objects.hx

```diff
@@ -103,11 +103,45 @@ tuple is the field name and the right value (_1) is the field value.
     );
 
 /**
+Determines whether an object has fields represented by a string path.  The path
+can contain object keys and array indices separated by ".".
+
+E.g. { key1: { key2: [1, 2, 3] } }.hasPath("key1.key2.2") -> returns true
+**/
+  public static function hasPath(o : {}, path : String, ?nonNullValue : Bool = false) : Bool {
+    var paths = path.split(".");
+    var current : Dynamic = o;
+
+    for (currentPath in paths) {
+      if(currentPath.isDigitsOnly()) {
+        var index = Std.parseInt(currentPath),
+            arr = Std.instance(current, Array);
+        if(null == arr || arr.length <= index) return false;
+        current = arr[index];
+      } else if (Reflect.hasField(current, currentPath)) {
+        current = Reflect.field(current, currentPath);
+      } else {
+        return false;
+      }
+    }
+    return nonNullValue ? current != null : true;
+  }
+
+/**
+Like `hasPath`, but will return `false` for null values, even if the key exists.
+
+E.g. { key1 : { key2: null } }.hasPathValue("key1.key2") -> returns false
+**/
+  public static function hasPathValue(o : {}, path : String) : Bool {
+    return hasPath(o, path, true);
+  }
```

Comment:
what if you remove the `nonNullValue` argument and change `hasPathValue` this way?
```
return getPath(o, path) != null;
```